hyperspy/drawing/markers.py

```diff
@@ -0,0 +1,830 @@
+            new_y_positions = new_y_positions + self.shift * yrange
+
+        new_kwds[key][..., 1] = new_y_positions
+
+        return new_kwds
+
+    def update(self):
+        if self._is_iterating or "relative" in [self._offset_transform, self._transform]:
+            kwds = self.get_data_position(get_static_kwargs=False)
+            self.collection.set(**kwds)
+
+    def _initialize_collection(self):
+        if self.collection is None:
+            self.collection = self._collection_class(
+                **self.get_data_position(),
+                offset_transform=self.offset_transform,
+            )
+            self.collection.set_transform(self.transform)
+
+    def plot(self, render_figure=True):
+        """
+        Plot a marker which has been added to a signal.
+
+        Parameters
+        ----------
+        render_figure : bool, optional, default True
+            If True, will render the figure after adding the marker.
+            If False, the marker will be added to the plot, but will the figure
+            will not be rendered. This is useful when plotting many markers,
+            since rendering the figure after adding each marker will slow
+            things down.
+        """
+        if self.ax is None:
+            raise AttributeError(
+                "To use this method the marker needs to be first add to a "
+                + "figure using `s._plot.signal_plot.add_marker(m)` or "
+                + "`s._plot.navigator_plot.add_marker(m)`"
+            )
+        self._initialize_collection()
+        self.collection.set_animated(self.ax.figure.canvas.supports_blit)
+        self.ax.add_collection(self.collection)
+        if render_figure:
+            self._render_figure()
+
+    def _render_figure(self):
+        self.ax.hspy_fig.render_figure()
+
+    def close(self, render_figure=True):
+        """Remove and disconnect the marker.
+
+        Parameters
+        ----------
+        render_figure : bool, optional, default True
+            If True, the figure is rendered after removing the marker.
+            If False, the figure is not rendered after removing the marker.
+            This is useful when many markers are removed from a figure,
+            since rendering the figure after removing each marker will slow
+            things down.
+        """
+        if self._closing:
+            return
+        self._closing = True
+        self.collection.remove()
+        self.events.closed.trigger(obj=self)
+        self._signal = None
+        for f in self.events.closed.connected:
+            self.events.closed.disconnect(f)
+        if render_figure:
+            self._render_figure()
+
+
+def is_iterating(arg):
+    return isinstance(arg, (np.ndarray, da.Array)) and arg.dtype == object
+
+
+def dict2vector(data, keys=None, return_size=True, dtype=float):
+    """Take some dictionary of values and create offsets based on the input keys.
```

Comment:
The docstring need to be updated and possibly tidying up now that the API of the new markers is settled? @CSSFrancis, can you please have a look and fix parsing the old API? Currently parsing arrows, rectangle, ellipse is not correct?

Reply:
@ericpre Yea, I'll look into it this afternoon!

Reply:
@ericpre I got a chance to look at this and we should be good(ish) now.  Unfortunately there are some things we can't replicate at the moment.  One of those is the `Quiver` class doesn't support quite as many of the fancy arrows that annotate used to so we can't plot an arrow with two arrows. 

It wouldn't be too hard of a change to the `Quiver` class to get the it so you could have two arrows but there are quite a few "fancy" options for arrows that would require some extra work.

Reply:
This is good as it is!